pkg/controllers/vdb/tlsconfig_reconciler.go

```diff
@@ -0,0 +1,193 @@
+			settlsconfig.WithNamespace(h.Vdb.GetObjectMeta().GetNamespace()),
+			settlsconfig.WithHTTPSTLSConfig(true),
+		}
+	} else {
+		opts = []settlsconfig.Option{
+			settlsconfig.WithClientServerTLSMode(h.Vdb.Spec.ClientServerTLSMode),
+			settlsconfig.WithClientServerTLSSecretName(h.Vdb.Spec.ClientServerTLSSecret),
+			settlsconfig.WithInitiatorIP(initiatorPod.GetPodIP()),
+			settlsconfig.WithNamespace(h.Vdb.GetObjectMeta().GetNamespace()),
+			settlsconfig.WithHTTPSTLSConfig(false),
+		}
+	}
+	return h.Dispatcher.SetTLSConfig(ctx, opts...)
+}
+
+func (h *TLSConfigReconciler) updateStatus(ctx context.Context) error {
+	var sRefs []*vapi.SecretRef
+	if h.TLSSecretType == vapi.HTTPSTLSSecretType {
+		sec := vapi.MakeHTTPSTLSSecretRef(h.Vdb.Spec.HTTPSNMATLSSecret)
+		if err1 := vdbstatus.UpdateSecretRef(ctx, h.VRec.GetClient(), h.Vdb, sec); err1 != nil {
+			return err1
+		}
+		sRefs = []*vapi.SecretRef{
+			sec,
+		}
+		h.Log.Info(fmt.Sprintf("Save secret %s into status", h.Vdb.Spec.HTTPSNMATLSSecret))
+	} else {
+		clientSec := vapi.MakeClientServerTLSSecretRef(h.Vdb.Spec.ClientServerTLSSecret)
+		if err3 := vdbstatus.UpdateSecretRef(ctx, h.VRec.GetClient(), h.Vdb, clientSec); err3 != nil {
+			return err3
+		}
+		sRefs = []*vapi.SecretRef{
+			clientSec,
+		}
+		h.Log.Info(fmt.Sprintf("Save secret %s into status", h.Vdb.Spec.ClientServerTLSSecret))
+	}
+	if err2 := vdbstatus.UpdateSecretRefs(ctx, h.VRec.GetClient(), h.Vdb, sRefs); err2 != nil {
+		return err2
+	}
+	var tlsModes []*vapi.TLSMode
+	if h.TLSSecretType == vapi.HTTPSTLSSecretType {
+		httpsTLSMode := vapi.MakeHTTPSTLSMode(h.Vdb.Spec.HTTPSTLSMode)
+		tlsModes = []*vapi.TLSMode{httpsTLSMode}
+		h.Log.Info(fmt.Sprintf("Save TLS mode %s into status", h.Vdb.Spec.HTTPSTLSMode))
+	} else {
+		clientTLSMode := vapi.MakeClientServerTLSMode(h.Vdb.Spec.ClientServerTLSMode)
+		tlsModes = []*vapi.TLSMode{clientTLSMode}
+		h.Log.Info(fmt.Sprintf("Save TLS mode %s into status", h.Vdb.Spec.ClientServerTLSMode))
+	}
+	err := vdbstatus.UpdateTLSModes(ctx, h.VRec.GetClient(), h.Vdb, tlsModes)
+	if err != nil {
+		h.Log.Error(err, "failed to update tls mode when setting up TLS")
+		return err
+	}
+	return nil
+}
+
+func (h *TLSConfigReconciler) checkIfTLSConfiguredInDB(ctx context.Context, initiatorPod *podfacts.PodFact) (bool, error) {
+	var sql string
+	if h.TLSSecretType == vapi.HTTPSTLSSecretType {
+		sql = "select certificate from tls_configurations where name='https';"
+	} else {
+		sql = "select certificate from tls_configurations where name='server';"
+	}
+	cmd := []string{"-tAc", sql}
+	stdout, stderr, err := h.PRunner.ExecVSQL(ctx, initiatorPod.GetName(), names.ServerContainer, cmd...)
+	if err != nil || strings.Contains(stderr, "Error") {
+		h.Log.Error(err, fmt.Sprintf("failed to check if TLS authentication is configured, stderr - %s", stderr))
+		return false, err
+	}
+	lines := strings.Split(stdout, "\n")
+	res := strings.Trim(lines[0], " ")
+	return res != "httpServerCert", nil
```

Comment:
I think it is safer to look for the specific tls config whose existence you are checking, in this case https_cert_*.
Are we 100% sure there's never a case certificate != httpServerCert while not being the certificate we are checking the existence for?